### PR TITLE
Increase Psalm to level 2

### PR DESCRIPTION
**Pull request by jordisala1991**
.

### CODE REVIEW

**Comment by jordisala1991** [comment]
Probably some of this should go to 1.x

File: src/Type/ImmutableArrayType.php [hunk 1]
```diff
@@ -50,7 +50,8 @@ public function configureOptions(OptionsResolver $resolver): void
             'keys' => [],
         ]);
 
-        $resolver->setAllowedValues('keys', static function ($value): bool {
+        $resolver->setAllowedTypes('keys', 'iterable');
```
Comment (jordisala1991) [inline src/Type/ImmutableArrayType.php]: Maybe would be nice to take a look at all types and try to narrow down the options, wdyt @VincentLanglet ?

Comment (VincentLanglet) [reply src/Type/ImmutableArrayType.php]: Indeed

Comment (jordisala1991) [reply src/Type/ImmutableArrayType.php]: I think it should be done i another PR , I open an issue.

Comment (jordisala1991) [inline src/Type/ImmutableArrayType.php]:
```suggestion
```

**Comment by jordisala1991** [comment]
The last psalm error makes no sense...

```
ERROR: PossibleRawObjectIteration - src/EventListener/ResizeFormListener.php:178:18 - Possibly undesired iteration over regular object ArrayAccess (see https://psalm.dev/208)
        foreach ($data as $name => $child) {
```

It looks like a bug to me, I have tried a lot of combinations and even adding an assert after those if does not make psalm happy

**Comment by jordisala1991** [comment]
> Probably some of this should go to 1.x

I answer to myself, yes it could make sense but a lot of errors are raised on 1.x due to deprecated code, I prefer to not invest time there.

**Comment by VincentLanglet** [comment]
> The last psalm error makes no sense...
> 
> ```
> ERROR: PossibleRawObjectIteration - src/EventListener/ResizeFormListener.php:178:18 - Possibly undesired iteration over regular object ArrayAccess (see https://psalm.dev/208)
>         foreach ($data as $name => $child) {
> ```
> 
> It looks like a bug to me, I have tried a lot of combinations and even adding an assert after those if does not make psalm happy

Changing
```
if (!\is_array($data) && !$data instanceof \Traversable) {
```
to 
`is_iterable` ?

Anyway, I didn't succeed to reproduce the error on psalm.dev ; can you ?
We could open an issue on psalm.

**Comment by jordisala1991** [comment]
I couldnt reproduce it either.

**Comment by VincentLanglet** [comment]
> I couldnt reproduce it either.

I succeed reproducing a similar issue https://psalm.dev/r/c95d52a172
We could try to add psalm-trace, I think it will be `ArrayAccess|array<array-key, mixed>`.

Seems like the issue is that the variable is not considered as `(Traversable&ArrayAccess)|array<array-key, mixed>`.

**Comment by jordisala1991** [comment]
To me it looks like a bug, should we suppress it for now?

**Comment by VincentLanglet** [comment]
> To me it looks like a bug, should we suppress it for now?

Sure, but we should open an issue on psalm and adding the link in the psalm-suppress IMHO.
Do you do it or should I do ?

**Comment by jordisala1991** [comment]
> > To me it looks like a bug, should we suppress it for now?
> 
> Sure, but we should open an issue on psalm and adding the link in the psalm-suppress IMHO. Do you do it or should I do ?

https://github.com/vimeo/psalm/issues/7928 

Here it is.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**